src/crystal/system/win32/socket.cr

```diff
@@ -254,11 +254,32 @@ module Crystal::System::Socket
   end
 
   private def system_reuse_port?
-    false
+    getsockopt_bool LibC::SO_REUSEADDR
   end
 
   private def system_reuse_port=(val : Bool)
-    raise NotImplementedError.new("Socket#reuse_port=")
+    if val
+      setsockopt_bool LibC::SO_EXCLUSIVEADDRUSE, false
+      setsockopt_bool LibC::SO_REUSEADDR, true
+    else
+      setsockopt_bool LibC::SO_REUSEADDR, false
+      setsockopt_bool LibC::SO_EXCLUSIVEADDRUSE, true
+    end
+    val
+  end
+
+  # and SO_REUSEADDR is always assumed on windows
+  # confusingly, the SO_REUSEADDR flag on windows is the equivalent of SO_REUSEPORT on linux
+  macro finished
+    # the address component of a binding can always be reused on windows
+    def reuse_address? : Bool
+      true
+    end
+
+    # there is no effect on windows
+    def reuse_address=(val : Bool)
+      val
+    end
   end
```

Comment:
I don't like this macro for overriding the definition in `Socket`. We should abstract the interface and delegate to platform-specific implementations instead.
Maybe it's okay to accept this for now and refactor afterwards, though?

Reply:
There are a bunch of other WebAssembly-specific definitions in `src/socket.cr` itself too. I'll take care of this later